### ISSUE TITLE
Allow production planning tool to include items without BOM and directly create a Material Request

### ISSUE THREAD

**Issue by gui-br** [frappe/erpnext#6664]
Sales with items without BOM are simply ignored when planning, I think that it should show an error on items without BOM, or create the production order as draft and validate the BOM on submit. Without some message that something went wrong getting sales orders items its difficult to know what was left on planning.

**Comment by Dbonejones** [comment]
Agreed, Sales items without BOM is an issue since items can then be missed if you use the PPT as your main planning tool.
Good changes to the PPT now that 7.1 has changed many things:

- ALL items (with BOM or without) MUST be considered
- "Default material request type" should be considered for the top level item as well (instead of defaulting to Production Order)

**Comment by bcornwellmott** [comment]
The Production Planning tool is exactly that, a **production** planning tool. If there is no BOM, it means there is no production happening. I'd suggest your first step after creating a sales order to be creating a material request from the sales order. From there you can use the production planning tool to process the Manufacture MRs, and create Supplier Quotations/Purchase orders for the Purchase MRs.

Where there is a disconnect is that there is no way to filter the MRs based on default MR type, thereby creating an MR for the purchase parts and another for the manufacture parts. You have to manually split them at the MR stage.

**Comment by Dbonejones** [comment]
Well, I think the production planning tool has grown beyond just planning production. Mainly from your work on it bcornwellmott. :)
People needed the PPT to produce MR's for Purchasing and Manufacturing because of how tied in Purchasing IS to Manufacturing in many businesses. You made some fantastic improvements that is making it possible for my business to use ERPNext.
The PPT is one of very few automation tools in ERPNext, and it's extremely useful for processing a Sales Order, planning production is only one part of that. Stock checking, and Purchasing is another part which the PPT already does quite well. Why not take it the rest of the way and allow top level items to be Purchase items, and not require a BOM? No downside that I can see.

I can work around the limitation easily now that I know it exists. But it seems worth posting an issue to consider a change in the future.

**Comment by rmehta** [comment]
@Dbonejones can you specify what message you want to show on what condition?

**Comment by bcornwellmott** [comment]
I think the best outcome would be to have an additional table in the PPT for Material requests. When the "Get Items" button is pressed (which loads the items with BOMs into the Production table), any directly purchased items from the Sales order get added to the new table. In addition, Instead of directly creating the Material Requests using the "Create Material Requests" button, you'd first be able to load all the Material Requests from the Production Orders in the list and review / edit them before submitting.

I hope this makes sense, but if not I can try to sketch it out with figures.

**Comment by Dbonejones** [comment]
I agree with bcornwellmott, an additional table in the PPT where you can review the MR's that could be created is a fantastic addition.
@rmehta If the PPT would look at all items on a Sales Order, even top level items with no BOM attached to them, and then look at the items "default material request type" for what kind of MR to create... I'm not sure if any additional messages need to be created?
